platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/Roster.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.roster;
+
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.platform.NodeId;
+import java.util.Collection;
+
+/**
+ * A roster is the set of nodes that are creating events and contributing to consensus.
+ */
+public interface Roster extends Iterable<RosterEntry>, SelfSerializable {
+
+    /**
+     * @return the nodeIds in the roster
+     */
+    Collection<NodeId> getNodeIds();
+
+    /**
+     * @return the RosterEntry with the given nodeId
+     * @throws java.util.NoSuchElementException if the nodeId is not in the roster
+     */
+    RosterEntry getEntry(NodeId nodeId);
+
```

Comment:
The following methods might be convenient:

-  `boolean contains(NodeId nodeId)`
- `int getSize()`


Something that probably requires further thought. Should we support lookup by index? E.g. get the 7th node in this roster? Perhaps something like `NodeId getNodeAtIndex(int index)`?

Reply:
no, I do not want to deal with indexes when we go to dynamic address book.  :-)  The definition of `7th node` changes when nodes are removed or added to the roster. 

Of course that means we can't use rosters as a drop-in replacement for address book.

Reply:
all algorithms will need to be migrated away from using index.

Reply:
Even if we don't do node index stuff, what about `contains()` and `getSize()`?

Reply:
Approving the PR for now, but I reserve the right to add these methods if they end up being convenient. ;)

Reply:
added with default implementations on the interface.